src/drawtypes/ramp.cpp

```diff
@@ -18,6 +18,11 @@ namespace drawtypes {
     return m_icons[math_util::cap<size_t>(index, 0, m_icons.size() - 1)];
   }
 
+  label_t ramp::get_by_percentage_with_borders(float percentage) {
+    size_t index = percentage * (m_icons.size() - 2) / 100.0f + 1;
```

Comment:
For small negative percentages (e.g. -1%), `index` will be 1 because the float is truncated towards 0. I'd suggest explicitly checking if the percentage `>= 100` or `<= 0` and returning `m_icons.size() -1` and `0` respectively.